src/util/options.schema.json

```diff
@@ -2125,6 +2117,12 @@
           "description": "Races with confidence at least threshold are warnings, lower are infos.",
           "type": "integer",
           "default": 0
+        },
+        "deterministic": {
+          "title": "warn.deterministic",
+          "description": "Output messages in deterministic order. Useful for cram testing.",
+          "type": "boolean",
+          "default": false
```

Comment:
How expensive is this sorting for larger benchmarks? If it is not excessive, we might just make it the default and avoid adding a further option.

Reply:
Message order is irrelevant in benchmarks though, so that would just be silly overhead to add.

The bigger reason to not make it default is usability. Right now messages are printed as transfer functions are evaluated, which happens in nice program order and in the order messages are created within a transfer function. It's actually quite nice to be able to read messages for linear code in the order of the code itself. Sorting them by text completely shuffles this up and makes `M.debug` things particularly useless since those often appear right before or after some meaningful messages.

Another reason is that I'm not confident of the reliability of this. This works fine for messages printed during transfer functions, but we also output messages from so many other stages of the analysis that some of them might not be properly accounted for. E.g. some messages may get created after one round of analysis and thus never get printed. This concerns incremental and server mode in particular.

Reply:
Fair points!